packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```diff
@@ -358,8 +369,40 @@ static void getExifFullInfo(ExifInterface exifInterface, Bundle response) {
     response.putParcelable("exif", exifMap);
   }
 
+  //  reference: https://developer.android.com/training/data-storage/shared/media#location-info-photos
+  @RequiresApi(api = Build.VERSION_CODES.Q)
+  static void getExifLocationForUri(ContentResolver contentResolver, Uri photoUri, Bundle asset) throws Exception {
+    // Get location data using the ExifInterface library.
+    // Exception occurs if ACCESS_MEDIA_LOCATION permission isn't granted.
+    photoUri = MediaStore.setRequireOriginal(photoUri);
+    InputStream stream = contentResolver.openInputStream(photoUri);
+    if (stream != null) {
+      ExifInterface exifInterface = new ExifInterface(stream);
```

Comment:
I think we should sync this somehow with https://github.com/expo/expo/pull/14408